django/forms/models.py

```diff
@@ -52,6 +52,11 @@ def construct_instance(form, instance, fields=None, exclude=None):
             continue
         if exclude and f.name in exclude:
             continue
+        # Leave defaults for fields that aren't in POST data, except for
```

Comment:
@timgraham This caused me a bit of a debugging headache. I expected that if I did something like this:

```
def clean(self):
    self.cleaned_data['some_field'] = 'some_value'
```

...that would be sufficient to set what was ultimately saved to the database. But I found that `some_field` was always being saved with its default value. It took me a while to figure out why, from looking at the Django source. Now I'm doing this, with success:

```
def __init__(self, *args, **kwargs):
    super(MyForm, self).__init__(*args, **kwargs)
    self.data = self.data.copy()
    self.data['some_field'] = 'some_value'
```

I'm just curious what the reasoning is behind the rule you implemented here?

Reply:
Your expectation seems reasonable. If there's a way to fix that while keeping backwards compatibility, feel free to offer a patch.

Reply:
Do Django's automated tests ensure backward compatibility? If not I probably don't have enough familiarity with Django to do so. Glad I'm not taking crazy pills, though. :)